test/unit/deferred.js

```diff
@@ -880,3 +875,27 @@ QUnit.test( "jQuery.when - chaining", function( assert ) {
 
 	defer.resolve( "other deferred" );
 } );
+
+QUnit.test( "jQuery.when - solitary thenables", function( assert ) {
+
+	assert.expect( 1 );
+
+	var rejected = new Promise( function( resolve, reject ) {
+		setTimeout( function() {
+			reject( "rejected" );
+		}, 100 );
+	} );
+
+	QUnit.stop();
```

Comment:
Yea, I started with that. I went with `QUnit.stop` given how prevalent it is in our code, but I'll switch it back.